src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp

```diff
@@ -531,7 +531,84 @@ struct MathFunctionName {
 
 template <>
 struct MathFunctionName<KrnlErfOp> {
-  static std::string functionName() { return "erff"; }
+  static std::string functionName(mlir::Type type) { return "erff"; }
```

Comment:
I would use the same template as you used, asserting an error if type is not float. If the erf is used for double format, that would be great too.